Sources/Signal.swift

```diff
@@ -24,8 +24,20 @@ public final class Signal<Value, Error: Swift.Error> {
 	/// when the signal terminates.
 	private var generatorDisposable: Disposable?
 
-	/// The state of the signal. `nil` if the signal has terminated.
-	private let state: Atomic<SignalState<Value, Error>?>
+	/// The status of the signal.
+	///
+ 	/// `status` synchronizes using Read-Copy-Update. Reads are thus wait-free,
+ 	/// but writes are required not to mutate in place. This suits `Signal` as
+ 	/// reads to `status` happens on the critical path of event delivery, while
+ 	/// adding and removing observers or termination generally has a constant
+	/// occurrence.
```

Comment:
(Packed) object references are the same size as the platform pointer size (4 or 8 bytes with the natural alignment), so reads (and writes that simply overwrite) are atomic as guaranteed by the architectures.

`statusLock` is introduced to serialise read-modify-write operations.